api/queries_sponsors.go

```diff
@@ -0,0 +1,90 @@
+package api
+
+import "net/http"
+
+type SponsorQuery struct {
+	User User `json:"user" graphql:"user(login: $username)"`
+}
+
+type User struct {
+	Sponsors Sponsors `json:"sponsors" graphql:"sponsors(first: 30)"`
+}
+
+type Sponsors struct {
+	TotalCount int    `json:"totalCount"`
+	Nodes      []Node `json:"nodes"`
+}
+
+type Node struct {
+	Login string `json:"login"`
+}
+
+func GetSponsorsList(httpClient *http.Client, hostname string, username string) ([]string, error) {
+	client := NewClientFromHTTP(httpClient)
+	//queryResult, err := querySponsorsViaReflection(client, hostname, username)
+	queryResult, err := querySponsorsViaStringManipulation(client, hostname, username)
+	if err != nil {
+		return nil, err
+	}
+
+	return mapQueryToSponsorsList(queryResult), nil
+}
+
+// This is currently broken. I suspect the issue is that the SponsorQuery
+// struct is not defining the variable $username correctly. I need to go
+// find an example query with a variable passed in like this to see how its
+// done. In the mean time, I'm going to move forward with the string
+// manipulation version of this function.
+func querySponsorsViaReflection(client *Client, hostname string, username string) (SponsorQuery, error) {
+	query := SponsorQuery{}
+	variables := map[string]interface{}{
+		"username": username,
+	}
+
+	err := client.Query(hostname, "SponsorsList", &query, variables)
+	if err != nil {
+		return SponsorQuery{}, err
+	}
+
+	return query, nil
+}
+
+func querySponsorsViaStringManipulation(client *Client, hostname string, username string) (SponsorQuery, error) {
+	type response struct {
+		SponsorQuery
+	}
+
+	query := `query SponsorsList($username: String!) {
+		user(login: $username) {
+			sponsors(first: 30) {
+				totalCount
+				nodes {
+					... on User {
+						login
+					}
+					... on Organization {
+						login
+					}
+				}
+			}	
+		}
+	}`
+	variables := map[string]interface{}{
+		"username": username,
+	}
+	var data response
+	err := client.GraphQL(hostname, query, variables, &data)
+	if err != nil {
+		return SponsorQuery{}, err
+	}
+
+	return data.SponsorQuery, nil
+}
+
+func mapQueryToSponsorsList(queryResult SponsorQuery) []string {
+	sponsorsList := []string{}
+	for _, node := range queryResult.User.Sponsors.Nodes {
+		sponsorsList = append(sponsorsList, node.Login)
+	}
+	return sponsorsList
+}
```

Comment:
```suggestion
func mapQueryToSponsorsList(queryResult SponsorQuery) []string {
	sponsorsList := make([]string, len(queryResult.User.Sponsors.Nodes))
	for i, node := range queryResult.User.Sponsors.Nodes {
		sponsorsList[i] = node.Login
	}
	return sponsorsList
}
```

Not a huge deal but since we already know the number of nodes we can construct a slice of the exact size rather than using `append` which might cause reallocations behind the scenes.

Reply:
Smart! I definitely need to improve my reflex to think about memory allocation... It's not something that you generally think about in TypeScript and Python 😅 I'd ask that you keep calling out these opportunities for me to help me build the muscle 💪